src/applications/personalization/profile360/actions/paymentInformation.js

```diff
@@ -94,11 +94,9 @@ export function savePaymentInformation(fields) {
     // };
 
     if (response.error || response.errors) {
-      recordEvent({
-        event: 'profile-edit-failure',
-        'profile-action': 'save-failure',
-        'profile-section': 'direct-deposit-information',
-      });
+      const errors = response?.error?.errors || [];
+      const analyticsData = createEventDataObjectWithErrors(errors);
```

Comment:
This is main purpose of this PR. The rest of the changes are to write the `createEventDataObjectWithErrors` helper.